pkg/volume/util.go

```diff
@@ -199,29 +202,39 @@ func (c *realRecyclerClient) WatchPod(name, namespace string, stopChannel chan s
 		return nil, err
 	}
 
-	eventCh := make(chan watch.Event, 0)
+	eventCh := make(chan watch.Event, 30)
 
 	go func() {
 		defer eventWatch.Stop()
 		defer podWatch.Stop()
 		defer close(eventCh)
-
+		var podWatchChannelClosed bool
+		var eventWatchChannelClosed bool
 		for {
 			select {
 			case _ = <-stopChannel:
 				return
 
 			case podEvent, ok := <-podWatch.ResultChan():
 				if !ok {
-					return
+					if eventWatchChannelClosed == true {
+						return
+					}
+					podWatchChannelClosed = true
+				} else {
+					podWatchChannelClosed = false
```

Comment:
When `_, ok <-podWatch.ResultChan() ` returns `ok == false`, can it ever happen that it returns `ok == true`? IMO that's not possible and therefore this `podWatchChannelClosed = false` is useless.